R/pin-read-write.R

```diff
@@ -57,12 +82,32 @@ vetiver_pin_write <- function(board, vetiver_model, ...) {
 
 #' @rdname vetiver_pin_write
 #' @export
-vetiver_pin_read <- function(board, name, version = NULL) {
+vetiver_pin_read <- function(board, name, version = NULL, check_renv = FALSE) {
 
     pinned <- pins::pin_read(board = board, name = name, version = version)
     meta   <- pins::pin_meta(board = board, name = name, version = version)
 
-    ## TODO: add subset of renv hash checking
+    if (check_renv) {
+        if (length(meta$user$renv_lock) > 0) {
+            local_lockfile <-
+                renv$snapshot(
+                    lockfile = NULL,
+                    packages = c(pinned$required_pkgs, "vetiver"),
+                    prompt = FALSE,
+                    force = TRUE
+                )
+            orig_lockfile <- structure(meta$user$renv_lock, class = "renv_lockfile")
+            renv_report_actions(local_lockfile, orig_lockfile)
+        } else {
+            cli::cli_warn(c(
+                "There is no lockfile stored with {.val {name}}:",
+                "i" = "Use {.arg check_renv = TRUE} when you save your model to your board"
+            ))
+        }
+    }
+
+    meta$user <- list_modify(meta$user, renv_lock = zap())
+    if (is_empty(meta$user)) names(meta$user) <- NULL
```

Comment:
This is needed because `list_modify()` returns an empty _named_ list here, not just `list()`.